src/talk/renderer/Viewer/viewer.composables.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { ref, watchEffect } from 'vue'
+import type { Ref } from 'vue'
+import { toValue } from '@vueuse/core'
+import type { MaybeRefOrGetter } from '@vueuse/core'
+import { fetchFileContent } from './viewer.service.ts'
+
+type FileContentComposableResult<T> = {
+	content: Ref<T>,
+	loading: Ref<boolean>,
+	error: Ref<boolean | string>,
+}
+
+export function useFileContent(filename: MaybeRefOrGetter<string>, format: 'binary'): FileContentComposableResult<Blob | null>
+export function useFileContent(filename: MaybeRefOrGetter<string>, format: 'text'): FileContentComposableResult<string | null>
+/**
+ * Fetch file content with reactive state in a watch effect
+ * @param filename - Path to user's file, e.g. '/Talk/file.txt'
+ * @param format - Format of the file content to be returned. Binary is returned as Blob
+ */
+export function useFileContent(filename: MaybeRefOrGetter<string>, format: 'text' | 'binary'): FileContentComposableResult<Blob | string | null> {
+	const content: Ref<Blob | string | null> = ref(null)
+	const loading: Ref<boolean> = ref(false)
+	const error: Ref<boolean | string> = ref(false)
+
+	watchEffect(async () => {
+		loading.value = true
+		content.value = null
+		error.value = false
+		try {
+			// Help TS to infer the type from overloads
+			content.value = format === 'text'
+				? await fetchFileContent(toValue(filename), format)
+				: await fetchFileContent(toValue(filename), format)
+		} catch (e) {
+			// TODO: We can parse error from e.response, but it is not translated
```

Comment:
I think a general line like `Could not load the file` would be enough (if error can be retrieved from console)

Reply:
Added `console.error`, thanks!